.github/workflows/test-build.yml

```
@@ -15,6 +15,58 @@ on:
+  set-build-info:
+    runs-on: ubuntu-latest
+    outputs:
+      branch: ${{ steps.build-info.outputs.branch }}
+      tag: ${{ steps.build-info.outputs.tag }}
+      trigger-sha: ${{ steps.build-info.outputs.trigger-sha }}
+      head-sha: ${{ steps.build-info.outputs.head-sha }}
+      pr-number: ${{ steps.build-info.outputs.pr-number }}
+    steps:
+      - name: Save build info (pull_request)
+        if: github.event_name == 'pull_request'
+        run: |
```

actionlint (1.7.7)

"github.event.pull_request.head.ref" is potentially untrusted. avoid using it directly in inline scripts. instead, pass it through an environment variable. see https://docs.github.com/en/actions/security-for-github-actions/security-guides/security-hardening-for-github-actions for more details

(expression)